.github/workflows/build.yml

```diff
@@ -97,3 +101,56 @@ jobs:
           meson setup build
           meson compile -C build
           meson test -C build
+
+  msys-meson:
+    name: 'msys-meson ${{ matrix.sys.abi }} ${{ matrix.library }}'
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: msys2 {0}
+    strategy:
+      matrix:
+        sys:
+          - { abi: mingw32, env: i686,         compiler: gcc }
+          - { abi: mingw64, env: x86_64,       compiler: gcc }
+          - { abi: ucrt64,  env: ucrt-x86_64,  compiler: gcc }
+          - { abi: clang32, env: clang-i686,   compiler: clang }
+          - { abi: clang64, env: clang-x86_64, compiler: clang }
+        library: ['shared', 'static']
+      fail-fast: false
+    steps:
+      - name: Use MinGW from MSYS
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.sys.abi}}
+          update: true
+          path-type: minimal
+          install: >-
+            mingw-w64-${{matrix.sys.env}}-toolchain
+            mingw-w64-${{matrix.sys.env}}-meson
+            mingw-w64-${{matrix.sys.env}}-ninja
+
+      - name: Setup compiler
+        if: startsWith(matrix.sys.abi, 'mingw') || startsWith(matrix.sys.abi, 'ucrt64')
+        run: |
+          CXX=${CC/#gcc/g++}
+          echo "CC=$CC" >> $GITHUB_ENV
+          echo "CXX=$CXX" >> $GITHUB_ENV
+        env:
+          CC: ${{ matrix.sys.compiler }}
+      - name: Setup compiler
+        if: startsWith(matrix.sys.abi, 'clang')
+        run: |
+          CXX=${CC/#clang/clang++}
+          echo "CC=$CC" >> $GITHUB_ENV
+          echo "CXX=$CXX" >> $GITHUB_ENV
+        env:
+          CC: ${{ matrix.sys.compiler }}
```

Comment:
Similarly, there is no need to mess w/ MSYS2 defaults, these are already set up accordingly, so the `compiler` variable can also be removed.